src/components/Profile/index.tsx

```diff
@@ -0,0 +1,32 @@
+'use client';
+
+import * as S from './style';
+
+import { RandomMentorImg, ProfileUpdateButton } from '@/components';
+import type { MentorType } from '@/types';
+
+interface Props {
+  profile: MentorType;
+}
+
+const Profile: React.FC<Props> = ({ profile: { name, generation } }) => {
+  const randomValue = Math.floor(Math.random() * 5); //0부터 4까지 중의 랜덤값 생성
```

Comment:
`Math.random()`을 위 방식으로 사용하게 되면, server 측과 client 측 둘 다 작동되어, 
렌더링 된 마크업이 일치하지 않게 되어 아래와 같이 hydrating 중 에러가 발생합니다.

`Uncaught Error: There was an error while hydrating. Because the error happened outside of a Suspense boundary, the entire root will switch to client rendering.`

해결 방법은 근본적으로 server side, client side 중 한 부분에서만 해당 코드가 실행되도록 해야하며,

1. useEffect 내부로 해당 코드를 옮긴다.
2. server 측에서 랜덤 값을 할당한다.
...

위 같은 방법들이 있습니다.

현재 메인페이지에서 server side 에서 각 요소에 랜덤 ImgNumber를 추가하고 있는 이유가 이것 때문입니다.
(1번 방식은 해당 imgNumber가 할당될 때까지 브라우저에서 빈 프로필 이미지가 보여지기 때문에 적용하지 않았습니다.)

일단 컴포넌트 제작 수준의 pr이니 이 상태로 pr을 마무리하고, 주석으로 추후 수정을 나타내는 TODO 주석을 달아놓으면 좋겠습니다

Reply:
예시를 들자면,

브라우저에서 보여지는 마크업
<img width="300" alt="스크린샷 2023-11-15 23 08 06" src="https://github.com/themoment-team/GSM-Networking-front/assets/80103328/a1c7c024-7fe2-43a0-84fa-a162f45cfe84">

서버에서 받아온 마크업
<img width="300" alt="스크린샷 2023-11-15 23 08 03" src="https://github.com/themoment-team/GSM-Networking-front/assets/80103328/7d615bec-3a97-4577-8102-a46115edcf26">

입니다

Reply:
자세한 설명 감사드립니다. 
[a3a5141](https://github.com/themoment-team/GSM-Networking-front/pull/30/commits/a3a5141f8490b8d30cd2fe8e1032341246e8f823) 주석 달아놓았습니다.

Reply:
확인 했습니다